packages/client/src/utils/index.ts

```diff
@@ -0,0 +1,33 @@
+import axios, { AxiosPromise } from "axios";
+import * as t from "io-ts";
+import { getOrElseW } from "fp-ts/Either";
+import { ClientError, getStatus } from "../models/ClientError";
+
+export function getAndValidateResponse<Expected>(
+  getRequest: AxiosPromise,
+  ressource: string,
+  decoder: t.Decoder<unknown, Expected>
+): Promise<Expected> {
+  return getRequest
+    .then((response) => {
+      return getOrElseW(() =>
+        Promise.reject(
+          `Unable to parse ${ressource}: ${JSON.stringify(response.data)}`
+        )
+      )(decoder.decode(response.data));
+    })
+    .catch((error: Error | string) => {
+      const status: number = typeof error == "string" ? 500 : getStatus(error);
+
+      const message: string =
+        axios.isAxiosError(error) && error.response
+          ? `Unable to retrieve ${ressource} with status code ${error.response.status}`
```

Comment:
I do agree, usually that's not the best.
Can be usefull in react tho.

I used it to avoid having big "ifs" with the same return statement at the end of each.
Do you want me to change it back ?